packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -910,21 +1066,38 @@ class Actions extends StatefulWidget {
   ) {
     assert(intent != null);
     assert(context != null);
-    Action<T>? action;
+    final Map<Action<T>, InheritedElement> actions = <Action<T>, InheritedElement>{};
     InheritedElement? actionElement;
+    final List<GlobalKey> registeredKeys = <GlobalKey>[];
 
-    _visitActionsAncestors(context, (InheritedElement element) {
-      final _ActionsMarker actions = element.widget as _ActionsMarker;
-      final Action<T>? result = actions.actions[intent.runtimeType] as Action<T>?;
+    bool searchActions(InheritedElement element, bool collect) {
+      final _ActionsMarker actionsMarker = element.widget as _ActionsMarker;
+      final Action<T>? result = actionsMarker.actions[intent.runtimeType] as Action<T>?;
       if (result != null) {
         actionElement = element;
         if (result.isEnabled(intent)) {
-          action = result;
-          return true;
+          actions[result] = element;
+          if (collect) {
+            // If we're collecting, and we find an action, we can stop looking.
+            return true;
+          }
         }
       }
+      if (collect && actionsMarker.registry != null) {
+        registeredKeys.addAll(actionsMarker.registry!._registeredActions);
+      }
       return false;
-    });
+    }
+
+    _visitActionsAncestors(context, (InheritedElement element) => searchActions(element, true));
+
+    if (actions.isEmpty) {
+      for (final GlobalKey key in registeredKeys) {
```

Comment:
Why do we need to walk the tree again from the widget that registered the actions? I thought `Actions.registerActions` means that only the actions provided by that particular Actions widget become available via the registry, but it looks like even the ancestor actions that didn't have registerActions set are made available? Is that intentional?

Reply:
In the current incarnation, I have it walking the tree, only looking at the registered actions, not all actions. I suppose we could just fetch the one from the registered context and only look there. I'll try that and see how it feels.

Reply:
OK, I modified it to only look at the one that the context points to, and not walk the tree at all.